pymodbus/datastore/store.py

```diff
@@ -51,7 +51,7 @@
 # ---------------------------------------------------------------------------#
 #  Datablock Storage
 # ---------------------------------------------------------------------------#
-class BaseModbusDataBlock:
+class BaseModbusDataBlock(dict):
```

Comment:
Why does it inherit from a dict, it really have nothing to do with a dict.

Reply:
And in most datastores the values are not stored in a dict.

Reply:
Inhering from dict would mean the following is a legal use:
```
   block = BaseModbusDataBlock()
   response = block["address"]
```
But that is not the use case, and never will be, since we never just get address, we always request address, count (using get/set/validate).

Reply:
The `block` dict is passed to the constructor for `ReactiveModbusslaveContext` using `**` unpacking.
https://github.com/pymodbus-dev/pymodbus/blob/031c4574eaf2c82995baab23e17ddaf9664057f2/pymodbus/server/reactive/main.py#L440-L446
That constructor takes several `BaseModbusDataBlock`s as keyword arguments.
https://github.com/pymodbus-dev/pymodbus/blob/031c4574eaf2c82995baab23e17ddaf9664057f2/pymodbus/server/reactive/main.py#L103-L116

This means that `block` must be something like `Dict[str, BaseModbusDataBlock]`

So any `block[item]` must also be `BaseModbusDataBlock`s, which means the two ways of creating an item must follow this:
https://github.com/pymodbus-dev/pymodbus/blob/031c4574eaf2c82995baab23e17ddaf9664057f2/pymodbus/server/reactive/main.py#L432-L438
(1) `block[modbus_entity]` = {... } fails because it's too broad (a dict).
(2)`block[modbus_entity] = db(...)` fails for because `db` is a too broad (also a dict).  If I type `db` to `BaseModbusDataBlock`, it cannot take the two arguments, anyways.

....somewhere around here I gave up, and made `BaseModbusDataBlock(dict)`. :)
Typing is very difficult; I will try again.

Reply:
You are again mixing issues....you talk about how the datastore is defined (__init__) and then by adding dict, you redefine the usage to be as a dict, which is wrong.

Please accept the 3 views of the datastore, that might help you quite a bit.